src/collects/seashell/backend/dispatch.rkt

```diff
@@ -451,7 +451,7 @@
         (_ _) ...)
        (define checksum
          (write-file project file (string->bytes/utf-8 contents)
-                     (if history (string->bytes/utf-8 history) "")
+                     (if history history "")
```

Comment:
It's probably easier to just leave history as bytes and convert here as your other code assumes
that history is a byte string (snapshots).